File: python/pc.py
from processor import processor
from disk import disk
from ram import ram

class pc():
    # atribut
    __pObj = processor()
    __dObj = disk()
    __rObj = ram()
    __totalPrice = 0

    # constructor
    def __init__(self):
        self.__pObj = processor()
        self.__dObj = disk()
        self.__rObj = ram()
        self.__totalPrice = 0
    
    def __init__(self, pObj = processor, dObj = disk, rObj = ram):
        self.__pObj = pObj
        self.__dObj = dObj
        self.__rObj = rObj

    # setter dan getter
    def setpObj(self, pObj):
        self.__pObj = pObj
        
    def getpObj(self):
        return self.__pObj 

    def setdObj(self, dObj):
        self.__dObj = dObj
        
    def getdObj(self):
        return self.__dObj 

    def setrObj(self, rObj):
        self.__rObj = rObj
        
    def getrObj(self):
        return self.__rObj 

    def setTotalPrice(self, totalPrice):
        self.__totalPrice = totalPrice
        
    def getTotalPrice(self):
        # self.__totalPrice =  self.pObj.getPrice() + self.dObj.getPrice() + self.rObj.getPrice()
        return self.__totalPrice
        # return self.__totalPrice 

File: java/pc.java
public class pc {
    // atribut serta nama objek
    private processor pObj;   
    private disk dObj;   
    private ram rObj;
    private int totalPrice;

    // constructor
    public pc() {
    }

    public pc(processor pObj, disk dObj, ram rObj) {
        this.pObj = pObj;
        this.dObj = dObj;
        this.rObj = rObj;
        this.totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice();
    }

    // setter dan getter
    public processor getpObj() {
        return pObj;
    }

    public void setpObj(processor pObj) {
        this.pObj = pObj;
    }

    public disk getdObj() {
        return dObj;
    }

    public void setdObj(disk dObj) {
        this.dObj = dObj;
    }

    public ram getrObj() {
        return rObj;
    }

    public void setrObj(ram rObj) {
        this.rObj = rObj;
    }

    public int getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(int totalPrice) {
        this.totalPrice = totalPrice;
    }   

    // untuk menampilkan output
    public void print(){
        System.out.println("\nProcessor");
        System.out.println("Nama : " + pObj.getName());
        System.out.println("Harga : " + pObj.getPrice());
        
        System.out.println("\nDisk");
        System.out.println("Type : " + dObj.getType());
        System.out.println("Kapasitas : " + dObj.getCapacity());
        System.out.println("Harga : " + dObj.getPrice());

        System.out.println("\nRAM");
        System.out.println("Kapasitas : " + rObj.getCapacity());
        System.out.println("Harga : " + rObj.getPrice());

    }
    
    
}
